sympy/physics/mechanics/kane.py

```diff
@@ -704,7 +704,10 @@ def kanes_equations(self, FL, BL):
             A list of all RigidBody's and Particle's in the system.
 
         """
-
+        if FL is None:
+            FL = [(self._inertial, 0*self._inertial.x)]
+        elif len(FL) == 0:
+            FL = [(self._inertial, 0*self._inertial.x)]
```

Comment:
This is still a hack. Look deeper into the code and figure out how to make the code handle empty iterables.